nav2_dwb_controller/dwb_plugins/src/standard_traj_generator.cpp

```diff
@@ -87,8 +90,8 @@ void StandardTrajectoryGenerator::initializeIterator(
 void StandardTrajectoryGenerator::checkUseDwaParam(
   const nav2_lifecycle::LifecycleNode::SharedPtr & nh)
 {
-  bool use_dwa;
-  nh->get_parameter_or("use_dwa", use_dwa, false);
+  bool use_dwa = false;
```

Comment:
The default value is set by the `declare_parameter` statement.

Reply:
Yes, that's true. However, get_parameter is not specifying a default value here. It is getting the value of the parameter and setting the use_dwa variable (which happens to be intialized to false). The value that is set depends on the default from declare_parameter and whether anyone has set the parameter.

Reply:
What I'm saying is there is no need to initialize use_dwa here because it's value is going to get set in the next line. Initializing it is just a wasted CPU cycle

Reply:
Ok, sure. Updated.